staging/src/k8s.io/client-go/util/workqueue/parallelizer.go

```diff
@@ -50,3 +51,32 @@ func Parallelize(workers, pieces int, doWorkPiece DoWorkPieceFunc) {
 	}
 	wg.Wait()
 }
+
+// ParallelizeUntilFeasible is a very simple framework that allow for parallelizing
+// N independent pieces of work until get feasible solution.
+func ParallelizeUntilFeasible(workers, pieces int, doWorkPiece DoWorkPieceFunc, feasible *int32) {
```

Comment:
The normal way to accomplish something like this in golang would be to use a `context.Context` for the work that you are dispatching, then cancelling the context when you want to abandon the rest of the work.

It would also make sense to have the workers fan in their results to a goroutine that handles results and decides when to cancel.

Reply:
Indeed, I understand what you mean, maybe a callback function can do the same, just like bsalamat said. And using Context may be too complicated for a public function.